software/react-native/concepts.md

```diff
@@ -0,0 +1,60 @@
+# Basic Concepts
+
+#### Request / Response Data Flow
+
+In a standard redux project data flows like this:
+
+1. Dispatch an `attempting` action that will attempt to `POST`/`GET` data from a server. This action will usually flip an `attempting: true` flag in it's domain-related reducer in case we want to display some sort of UI to demonstrate to the user that a request is in process.
```

Comment:
Might be worth noting that we don't always have a state slice related to `attempting: true` for every single API call, certain cases we may have a single state like `isLoading: true` or something. Of course it just depends upon what we degree we are trying to display on the UI.

This is certainly a common practice though for sure, perhaps we should make more of a convention around it